google-cloud-translate/src/main/java/com/google/cloud/translate/TranslateOptions.java

```diff
@@ -135,7 +136,19 @@ public TranslateOptions build() {
 
   private TranslateOptions(Builder builder) {
     super(TranslateFactory.class, TranslateRpcFactory.class, builder, new TranslateDefaults());
-    this.apiKey = builder.apiKey != null ? builder.apiKey : getDefaultApiKey();
+    if (builder.apiKey != null) {
+      this.apiKey = builder.apiKey;
+    } else if (System.getenv(ServiceOptions.CREDENTIAL_ENV_NAME) == null) {
+      this.apiKey = getDefaultApiKey();
+    } else {
```

Comment:
Again, I think you have this wrong -- the only time that the logger.log will happen is when apiKey hasn't been set && you have a service account (ie. GOOGLE_APPLICATION_CREDENTIALS is set)